tensorflow/python/eager/function.py

```diff
@@ -2639,46 +2629,46 @@ def canonicalize_function_inputs(self, *args, **kwargs):
               "{} got keyword argument `{}` that was not included in "
               "input_signature".format(self.signature_summary(), arg))
 
+    num_req_args = len(self._arg_names) - len(self._fullargspec.defaults or [])
```

Comment:
you probably can make this as a class variable, instead of calculating it every time.

Reply:
Good point